cli/src/lib/configs/initConfig.ts

```diff
@@ -415,38 +421,96 @@ export function getReadonlyConfigInitFunction<
+      // eslint-disable-next-line @typescript-eslint/consistent-type-assertions
+      latestConfig = config as LatestConfig;
     } else {
-      latestConfig = await ensureConfigIsValidLatest({
-        config,
-        configPath,
-        validateLatestConfig,
-        validate,
-        latestConfigVersion,
-      });
+      if (
+        typeof config !== "object" ||
+        config === null ||
+        !("version" in config) ||
+        (typeof config.version !== "number" &&
+          typeof config.version !== "string")
+      ) {
+        return commandObj.error(
+          `Invalid config at ${color.yellow(
+            configPath,
+          )}. Expected to have a ${color.yellow("version")} property of type number`,
+        );
+      }
+
+      const currentConfigVersion = Number(config.version);
+
+      if (!isInteger(currentConfigVersion) || currentConfigVersion < 0) {
+        return commandObj.error(
+          `Invalid config at ${color.yellow(
+            configPath,
+          )}. ${color.yellow("version")} property should be a positive integer. Got: ${color.yellow(
+            config.version,
+          )}`,
+        );
+      }
+
+      const currentSchema = allSchemas[currentConfigVersion];
+
+      if (currentSchema === undefined) {
+        return commandObj.error(
+          `Invalid config at ${color.yellow(
+            configPath,
+          )}. ${color.yellow("version")} property should be less than or equal to ${color.yellow(
+            allSchemas.length - 1,
+          )}. Got: ${color.yellow(config.version)}. Make sure you are using a correct version of the CLI that supports this config version`,
+        );
+      }
+
+      const validateCurrentConfigVersion = addFormats(
+        new Ajv.default({
+          allowUnionTypes: true,
+        }),
+      ).compile<Config>(currentSchema);
+
+      if (!validateCurrentConfigVersion(config)) {
+        await updateSchemaPathInConfigString(
+          validateCurrentConfigVersion.schema,
+        );
+
+        return commandObj.error(
+          `Invalid config at ${color.yellow(
+            configPath,
+          )}. ${await validationErrorToString(
+            validateCurrentConfigVersion.errors,
+          )}${
+            currentConfigVersion === latestConfigVersion
+              ? ""
+              : `\n\nConfig version is ${color.yellow(currentConfigVersion)}, while the latest config version (that ${CLI_NAME_FULL} of version ${commandObj.config.version} is aware of) is ${color.yellow(
+                  latestConfigVersion,
+                )}. ${CLI_NAME_FULL} will automatically migrate your config to the latest version when you fix validation errors (that are listed above) and run any command that uses this config`
+          }`,
+        );
+      }
+
+      if (Number(config.version) < migrations.length) {
+        ({ latestConfig, configString } = await migrateConfig({
```

Comment:
why in braces?

Reply:
this one is the only way to destructure-reassign in js https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment